ostd/src/arch/x86/trap/syscall.S

```diff
@@ -62,23 +62,29 @@ syscall_return:
     # trap_num
     # error_code
 
-    # determain sysret or iret
-    cmp dword ptr [rsp + 4*8], 0x100  # syscall?
-    je sysret
-iret:
-    # construct trap frame
-    push {USER_SS}          # push ss
-    push [rsp - 8*8]        # push rsp
-    push [rsp + 3*8]        # push rflags
-    push {USER_CS}          # push cs
-    push [rsp + 4*8]        # push rip
-
-    iretq
+    # Determine whether to use sysret or iret.
+    # If returning to user space with a clean context,
+    # the fast sysret path can be used;
+    # otherwise, the slower iret path should be used.
+    # Reference: <https://elixir.bootlin.com/linux/v6.0.9/source/arch/x86/entry/entry_64.S#L122>.
+    cmp qword ptr [rsp], rcx # sysret requires rcx = rip
+    jne iret
+    cmp qword ptr [rsp + 8], r11 # sysret requires r11 = rflags
+    jne iret 
+    test r11, 0x10100 # sysret requires rflags not contain RF and TF flags
```

Comment:
Try to 
 - have at least two spaces before `# ...`.
 - align all `# sysret requires ...` comments.

```
    cmp qword ptr [rsp], rcx      # sysret requires rcx = rip
    jne iret

    cmp qword ptr [rsp + 8], r11  # sysret requires r11 = rflags
    jne iret

    test r11, 0x10100             # sysret requires rflags not contain RF and TF flags
    jne iret

    shl rcx, 64 - {ADDRESS_WIDTH}
    sar rcx, 64 - {ADDRESS_WIDTH}
    cmp qword ptr [rsp], rcx      # sysret requires rip be a canonical address
    je sysret
    mov rcx, [rsp]

iretd:
```